src/components/RegistrationForm.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react'
+
+
+export default class RegistrationForm extends Component {
+  constructor( props) {
+    super ( props )
+    this.state = {
+      name: '',
+      email: '',
+      password: '',
+      phone_number: '',
+      company: ''}
+  }
+
+  handleInput = ( event, stateKey ) => {
+    this.setState({
+      name: event.target.fullName,
+      email: event.target.email,
+      password: event.target.password,
+      phone_number: event.target.number,
+      company: event.target.company,
+      newsletter_subscribed: true
+     })
+  }
+
+  render() {
+    return (
+      <form action="http://localhost:5000/api/v1/users/register" method="post">
+        <input
+          type="text"
+          name="name"
+          placeholder="Name"
+          value={this.state.name}
+          onChange={this.handleInput} />
+        <input
+          type="text"
+          name="email"
+          placeholder="Company email address"
+          defaultValue={this.props.location.query.email}
```

Comment:
We should be passing in the props that are used by the component - this usage implies that the remainder of the location object is not used, and that we should just pass in an email prop?